.github/CONTRIBUTING.md

```diff
@@ -34,151 +30,191 @@ ensure that the community is free to use your contributions.
+
+**Terra Add-on**
+
+For the former, we just need to call the `setup.py` script:
+
+    qiskit-aer$ python ./setup.py bdist_wheel
+
+We are using **scikit-build** as a substitute for **setuptools**. This is
+basically the glue between **setuptools** and **CMake**, so there are various
+options to pass variables to **CMake**, and the underlying build system
+(depending on your platform). The way to pass variables is:
+
+    qiskit-aer$ python ./setup.py bdist_wheel -- -DCMAKE_VARIABLE=Values -- -
+    Makefile_Flag
+
+So a real example could be:
+
+    qiskit-aer$ python ./setup.py bdist_wheel -- -j8
+
+This is setting the **CMake** variable `STATIC_LINKING` to value `True` so
+**CMake** will try to create an statically linked **cython** library, and is
+passing `-j8` flag to the underlaying build system, which in this case is
+Makefile, telling it that we want to build in parallel, using 8 processes.
+
+**Standalone Executable**
+
+If we want to build a standalone executable, we have to use **CMake** directly.
+The preferred way **CMake** is meant to be used, is by setting up an "out of
+source" build. So in order to build our standalone executable, we have to follow
+these steps:
+
+    qiskit-aer$ mkdir out
+    qiskit-aer$ cd out
+    qiskit-aer/out$ cmake ..
+    qiskit-aer/out$ cmake --build . --config Release -- -j4
+
+Once built, you will have your standalone executable into the `Release/` or
+`Debug/` directory (depending on the type of building chosen with the `--config`
+option):
 
-**Windows**
+    qiskit-aer/out$ cd Release
+    qiskit-aer/outRelease/$ ls
+    aer_simulator_cpp
 
-On Windows you must have Anaconda3 installed in the system, and We recommend installing
-Visual Studio 2017 (Communit Edition).
-The same rules applies when searching for an OpenBLAS implementation, if CMake can't
-find one suitable implementation installed in the system, it will take the BLAS
-library from the Anaconda3 environment.
 
 
+### macOS
 
+There are various methods depending on the compiler we want to use. If we want
+to use the **Clang** compiler, we need to install an extra library for
+supporting **OpenMP**: **libomp**. The **CMake** build system will warn you
+otherwise. To install it manually, in a terminal window, run:
 
-### Building
+    brew install libomp
 
+We recommend installing **OpenBLAS**, which is our default choice:
 
-There are two ways of building Aer simulators, depending on our goal they are:
-1. Build Terra compatible addon.
-2. Build standalone executable
+    brew install openblas
 
-**Terra addon**
+The **CMake** build system will search for other **BLAS** implementation
+alternatives if **OpenBLAS** is not installed in the system.
+
+You further need to have **Xcode Command Line Tools** installed on macOS:
+
+    xcode-select --install
+
+There are two ways of building Aer simulators, depending on your goal:
+
+1. Build a Terra compatible add-on;
+2. Build a standalone executable.
+
+**Terra Add-on**
```

Comment:
Are we repeating the same instructions here?
One of the  cool things of using `scikit-build / CMake` is that we always build in the same way. One build system for all platforms.

Reply:
I was totally confused by the mixture of linux, macOS, windows caveats in the original version of the doc, so I figured separating the platforms out would make reading easier. I'll try to combine them and maintain clarity.

Reply:
I agree with Paul, seperate sections and repetition is much clearer. That way you can just jump to one section and read everything not have to go back and forth fishing for cmake flags and dependencies